packages/bitcore-wallet-client/src/lib/common/utils.ts

```diff
@@ -503,4 +503,21 @@ export class Utils {
       return { uncheckedSerialize: () => unsignedTxs };
     }
   }
+
+  static getCurrencyCodeFromCoinAndChain(coin: string, chain: string): string {
```

Comment:
I think this can be cleaned up a bit for readability.
```
 CHAINSUFFIXMAP: {
    e: 'eth',
    m: 'matic'
  }

...

const matchingSuffixEntry = Object.entries(Constants.CHAINSUFFIXMAP).find(
      ([_, chainCode]) => chain.toLowerCase() === chainCode
    ) as [string, string];
    const suffix = matchingSuffixEntry && matchingSuffixEntry[0];
    const coinIsAnotherChain = Object.values(Constants.CHAINSUFFIXMAP).find(
      chainCode => chainCode === coin.toLowerCase()
    );
```
can be
```
 CHAINSUFFIXMAP: {
    eth: 'e',
    matic: 'm'
}

...

const suffix = Constants.CHAINSUFFIXMAP[chain.toLowerCase()];
const coinIsAChain = !!Constants.CHAINSUFFIXMAP[coin.toLowerCase()]; // e.g. MATIC ERC20 token
```